### PR TITLE
Adding Bigtable RowFilter base class.

### PR DESCRIPTION
**Pull request by dhermes**
Also adding regex filters for a row key (bytes) and a family name (string).

These come from

https://github.com/GoogleCloudPlatform/cloud-bigtable-client/blob/6a498cd3e660c7ed18299e980c1658d67661e69b/bigtable-protos/src/main/proto/google/bigtable/v1/bigtable_data.proto#L321-L404

In addition to more classes for primitive properties, parent classes are forthcoming to handle the non-primitive cases of filter Chain, Interleave and Condition (ternary).

Also renaming some redunant unit test names in `test_column_family.py` and ditching use of `NotImplementedError` in bigtable base classes (for both GC rule and row filter).

### CODE REVIEW

**Comment by dhermes** [comment]
@tseaver As with #1290 I am very open to breaking this class with 13 possible filters into 13 separate classes, but wanted to get the pattern in full view before we make a decision in #1290.

**Comment by dhermes** [comment]
@tseaver I'm going to start with just two here.

**Comment by dhermes** [comment]
@tseaver PTAL. I opted not to have regex filters inherit from a parent, but these ended up having also identical unit tests, so maybe it would've been worth it? I figured it was unnecessary since the classes to so little. (FWIW of the 19 filters, 4 of them are regex filters)

**Comment by tseaver** [comment]
If four of the filters take only a `regex` ctor argument, and differ only in terms of how they construct the protobuf, then factoring out `__init__`, `__eq__`, and `__ne__` into a base class seems like a win.

**Comment by dhermes** [comment]
OK. Will re-factor. `__ne__` is already in the main base class, but the other 2 are still uniform. Will ping once done.

**Comment by dhermes** [comment]
@tseaver PTAL. I added `_RegexFilter` as a base class.

File: gcloud/bigtable/row.py [hunk 1]
```diff
@@ -31,3 +32,111 @@ class Row(object):
...
+
+    def __init__(self, regex):
+        self.regex = regex
+
+    def __eq__(self, other):
+        if not isinstance(other, self.__class__):
+            return False
+        return other.regex == self.regex
+
+    def to_pb(self):
```
Comment (dhermes) [inline gcloud/bigtable/row.py]: This re-implementation is a "sacrifice to the `pylint` god" as @tseaver would say.

It wants all virtual methods to be implemented by subclasses. I could also just remove `to_pb` from both bases `RowFilter` and `_RegexFilter`. `NotImplementedError` is not really useful anyhow

Comment (tseaver) [inline gcloud/bigtable/row.py]: +1 to dropping it in both places.

**Comment by dhermes** [comment]
@tseaver PTAL I dropped all the virtual `to_pb` methods.

**Comment by tseaver** [comment]
LGTM

**Pull request merged**